Classes/Controller/BackendServiceController.php

```diff
@@ -393,25 +400,51 @@ public function loadTreeAction(NodeTreeBuilder $nodeTreeArguments, $includeRoot
     /**
      * @throws \Neos\Flow\Mvc\Exception\NoSuchArgumentException
      */
-    public function initializeGetPolicyInformationAction()
+    public function initializeGetLazyNodeDataAction()
     {
         $this->arguments->getArgument('nodes')->getPropertyMappingConfiguration()->allowAllProperties();
     }
 
     /**
+     * Fetches all the node information that can be lazy-loaded
+     *
      * @param array<NodeInterface> $nodes
      */
-    public function getPolicyInformationAction(array $nodes)
+    public function getLazyNodeDataAction(array $nodes)
     {
         $result = [];
         /** @var NodeInterface $node */
         foreach ($nodes as $node) {
-            $result[$node->getContextPath()] = ['policy' => $this->nodePolicyService->getNodePolicyInformation($node)];
+            $otherNodeVariants = array_values(array_filter(array_map(function ($node) {
+                return $this->getCurrentDimensionPresetIdentifiersForNode($node);
+            }, $node->getOtherNodeVariants())));
+            $result[$node->getContextPath()] = [
+                'policy' => $this->nodePolicyService->getNodePolicyInformation($node),
+                'dimensions' => $this->getCurrentDimensionPresetIdentifiersForNode($node),
+                'otherNodeVariants' => $otherNodeVariants
```

Comment:
Wait so this contains only dimension presets in which variants exist? not even information about those variants. It seems to me super expensive to do this for every node just to get the "filled presets".

Reply:
What kind of information could we load here? We already have dimension presets information in the UI. That's really all we need.
Is there cheaper way to get this info?